transport_nantes/topicblog/views.py

```diff
@@ -305,25 +323,28 @@ class TopicBlogItemList(TopicBlogBaseList):
 
     def get_template_names(self):
         names = super().get_template_names()
-        if 'item_slug' in self.kwargs:
+        if 'the_slug' in self.kwargs:
             return ['topicblog/topicblogitem_list_one.html'] + names
         else:
             return names
 
 
-@StaffRequired
-def get_slug_dict(request):
-    """Return a list of all existing slugs"""
-    qs = TopicBlogItem.objects.order_by('slug').values('slug')
-    dict_of_slugs = Counter([item['slug'] for item in qs])
-    return JsonResponse(dict_of_slugs, safe=False)
+######################################################################
+# TopicBlogEmail
 
+class TopicBlogEmailEdit(TopicBlogBaseEdit):
```

Comment:
The get_context_data from TBOB doesn't define context["form_admin"], context["form_content_a"], context["form_content_b"] and context["form_social"], but it's needed to display the form properly. 
It's been defined inside TBITem, but surely needs to be added on TBEmail as well.

Reply:
Might that not vary from one object type's editor to the next?

I'm inclined to go for duplication until we understand better what is common.  There's a fine line between abstracting too early and too late, but it has something to do with how far ahead one can clearly see.

Reply:
I agree that it shouldn't be defined on the TBOB level, what I was pointing is its absence on TBEmail level, if you go to tb/admin/e/new with this configuration, you will have 4 empty tabs. No field will show.
A difference you introduced between TBI and TBE is for example the body_image_2. It doesn't exist in TBI, but does in TBE. 
Is it something you intended ?

Reply:
Yes, although this commit makes no claim that the views do anything at all.   If you are testing this at 607a2d1 (`origin/TBEmail-design-pedagical`), then it works a bit better, but the point of that branch is get the email views working.

What do you think should happen on this in this commit?

Reply:
Alright, in that optic indeed my comment is a bit out of scope. It does 200 and if it's what's the intent of this commit is, it works :slightly_smiling_face: